### PR TITLE
chacha20 performance optimizations for ppc64le specifically for Power10.

### PR DESCRIPTION
**Pull request by dannytsen**
Performance increase around 50%.

Co-authored-by: Madhusudhanan Duraisamy <madurais@in.ibm.com>

Co-authored-by: Nilamjyoti Goswami <nilamgoswami@in.ibm.com>

Co-authored-by: Siva Sundar Anbareeswaran <srisivasundar@in.ibm.com>

Reviewed-by: Danny Tsen <dtsen@us.ibm.com>
Tested-by: Danny Tsen <dtsen@us.ibm.com>
Signed-off-by: Danny <dtsen@us.ibm.com>

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by dannytsen** [comment]
Hi, I have my CLA cleared and upstreamed the patch for the actual author at IBM.  Deepankar is added as Author to ensure the credits.  I signed off the patch as needed.

**Comment by mattcaswell** [comment]
All authors and co-authors need to have an ICLA registered and be mentioned in schedule A of the IBM CCLA.

File: crypto/chacha/asm/chachap10-ppc.pl [hunk 1]
```diff
@@ -0,0 +1,1379 @@
...
+
+sub VSX_lane_ROUND_1x {
+
+my $a=@_[0];
+my $b=@_[1];
+my $c=@_[2];
+my $d=@_[3];
+my $odd=@_[4];
+     
+     	vadduwm		($a,$a,$b);
```
Comment (bernd-edlinger) [inline crypto/chacha/asm/chachap10-ppc.pl:112]: there is trailing whitespace all over this file
and also blank before tab.

File: crypto/chacha/asm/chachap10-ppc.pl [hunk 2]
```diff
@@ -0,0 +1,1379 @@
...
+my $odd=@_[4];
+     
+     	vadduwm		($a,$a,$b); 
+	vxor		($d,$d,$a);
+	vrlw		($d,$d,$sixteen);  
+
+
+	vadduwm		($c,$c,$d);
+	vxor		($b,$b,$c);
+	vrlw		($b,$b,$twelve)  ;
```
Comment (bernd-edlinger) [inline crypto/chacha/asm/chachap10-ppc.pl:119]: and why is there a space before the semiolon?

**Review by bernd-edlinger — COMMENTED** [review]
just some nits

**Comment by mattcaswell** [comment]
Ping @t8m

**Comment by t8m** [comment]
@dannytsen could you please fix the issues mentioned by Bernd?

**Comment by dannytsen** [comment]
Hi @t18m, I'll relay to the authors.  Thanks.

**Comment by dannytsen** [comment]
@t8m fixed trailing spaces.  Please review.  Thanks.

**Comment by dannytsen** [comment]
@t8m Could you do the review the fixes on trailing spaces?  Thanks.

**Review by t8m — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
Merged with trailing whitespace fixes.  Thanks for this contribution.

**Pull request closed without merging**